### PR TITLE
Add support for linp.sensor_occupy.es3

### PR DESCRIPTION
**Pull request by kei81131**
https://home.miot-spec.com/spec/linp.sensor_occupy.es3

image

### CODE REVIEW

File: custom_components/xiaomi_gateway3/core/devices.py [hunk 1]
```diff
@@ -1660,6 +1660,19 @@
...
     ],
+}, {
+    # https://home.miot-spec.com/spec/linp.sensor_occupy.es3
+    20731: ["Linp", "Occupancy Sensor", "ES3", "linp.sensor_occupy.es3"],
+    "spec": [
+        # main sensors
+        BoolConv("occupancy", "binary_sensor", mi="2.p.1078"),
+        BaseConv("illuminance", "sensor", mi="2.p.1005"),
+        # other sensors
+        BaseConv("battery", "sensor", mi="3.p.1003"),
```
Comment (CharmingCheung) [inline custom_components/xiaomi_gateway3/core/devices.py:1671]:
```suggestion
        BaseConv("battery", "sensor", mi="4.p.1003", entity=ENTITY_LAZY),
```

File: custom_components/xiaomi_gateway3/core/devices.py [hunk 2]
```diff
@@ -1660,6 +1660,19 @@
         BaseConv("has_someone_duration", "sensor", mi="2.p.1081", entity=ENTITY_DISABLED),
         BaseConv("no_one_duration", "sensor", mi="2.p.1082", entity=ENTITY_DISABLED),
     ],
+}, {
+    # https://home.miot-spec.com/spec/linp.sensor_occupy.es3
+    20731: ["Linp", "Occupancy Sensor", "ES3", "linp.sensor_occupy.es3"],
```
Comment (CharmingCheung) [inline custom_components/xiaomi_gateway3/core/devices.py:1665]:
```suggestion
    20731: ["Linptech", "Human Presence Sensor ES3", "ES3BB", "linp.sensor_occupy.es2"],
```

File: custom_components/xiaomi_gateway3/core/devices.py [hunk 3]
```diff
@@ -1660,6 +1660,19 @@
         BaseConv("has_someone_duration", "sensor", mi="2.p.1081", entity=ENTITY_DISABLED),
         BaseConv("no_one_duration", "sensor", mi="2.p.1082", entity=ENTITY_DISABLED),
     ],
+}, {
+    # https://home.miot-spec.com/spec/linp.sensor_occupy.es3
```
Comment (CharmingCheung) [inline custom_components/xiaomi_gateway3/core/devices.py:1664]:
```suggestion
    # https://home.miot-spec.com/spec/linp.sensor_occupy.es3
    # https://home.mi.com/views/introduction.html?model=linp.sensor_occupy.es2
```

**Review by CharmingCheung — CHANGES_REQUESTED** [review]
Linptech ES3 is `linp.sensor_occupy.es2`. You can check the instruction manual at this URL: https://home.mi.com/views/introduction.html?model=linp.sensor_occupy.es2

**Comment by AlexxIT** [comment]
Thanks!

**Comment by AlexxIT** [comment]
https://github.com/AlexxIT/XiaomiGateway3/releases/tag/v4.0.8

**Pull request merged**